From a176fe662fc7c99bd9229dddae13b50679593f66 Mon Sep 17 00:00:00 2001
From: "po.cheng" <po.cheng@adlinktech.com>
Date: Fri, 29 Dec 2023 09:58:59 +0800
Subject: [PATCH 09/22] sp2-imx8mp: patch: pwm_backlight: work around to fix
 lvds display: enable pwm again at the end of set brightness

Signed-off-by: po.cheng <po.cheng@adlinktech.com>
---
 drivers/video/pwm_backlight.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/video/pwm_backlight.c b/drivers/video/pwm_backlight.c
index d7c096923b37..28cc74568fcd 100644
--- a/drivers/video/pwm_backlight.c
+++ b/drivers/video/pwm_backlight.c
@@ -184,6 +184,11 @@ static int pwm_backlight_set_brightness(struct udevice *dev, int percent)
 				return log_ret(ret);
 		}
 		priv->enabled = false;
+	} else {
+		ret = pwm_set_enable(priv->pwm, priv->channel, true);
+		if (ret)
+			return log_ret(ret);
+		priv->enabled = true;
 	}
 
 	return 0;
-- 
2.25.1

